bokeh/models/widgets/styles.py

```diff
@@ -0,0 +1,115 @@
+    border_top_style = BorderStyle
+    border_top_width = BorderWidth
+
+    border_right_color = Or(Color, Transparent, default=None)
+    border_right_style = BorderStyle
+    border_right_width = BorderWidth
+
+    border_bottom_color = Or(Color, Transparent, default=None)
+    border_bottom_style = BorderStyle
+    border_bottom_width = BorderWidth
+
+    border_left_color = Or(Color, Transparent, default=None)
+    border_left_style = BorderStyle
+    border_left_width = BorderWidth
+
+    top    = Or(Length, Percentage, Auto, default=None)
+    right  = Or(Length, Percentage, Auto, default=None)
+    bottom = Or(Length, Percentage, Auto, default=None)
+    left   = Or(Length, Percentage, Auto, default=None)
+
+    margin        = Tupled_1_4(Or(Length, Percentage, Auto), default=None)
+    margin_top    = Or(Length, Percentage, Auto, default=None)
+    margin_right  = Or(Length, Percentage, Auto, default=None)
+    margin_bottom = Or(Length, Percentage, Auto, default=None)
+    margin_left   = Or(Length, Percentage, Auto, default=None)
+
+    padding        = Tupled_1_4(Or(Length, Percentage), default=None)
+    padding_top    = Or(Length, Percentage, default=None)
+    padding_right  = Or(Length, Percentage, default=None)
+    padding_bottom = Or(Length, Percentage, default=None)
+    padding_left   = Or(Length, Percentage, default=None)
+
+    color = Color
+
+    float = Enum("left", "right", "none", "inline-start", "inline-end", default=None)
+    clear = Enum("none", "left", "right", "both", "inline-start", "inline-end", default=None)
+
+    font_family = String
+    font_size = Or(Enum("xx-small", "x-small", "small", "medium", "large", "x-large", "xx-large"), Enum("larger", "smaller"), Length, Percentage, default=None)
+    font_style = Enum("normal", "italic", "oblique", default=None)
+    font_weight = Enum("normal", "bold", "bolder", "lighter", "100", "200", "300", "400", "500", "600", "700", "800", "900", default=None)
+
+    line_height = Or(Enum("normal"), Float, Int, Length, Percentage, default=None)
+
+    overflow   = Or(Enum("visible", "hidden", "scroll"), Auto, default=None)
+    overflow_x = Or(Enum("visible", "hidden", "scroll"), Auto, default=None)
+    overflow_y = Or(Enum("visible", "hidden", "scroll"), Auto, default=None)
+
+    text_align = Enum("start", "end", "left", "right", "center", "justify", "match-parent", default=None)
+    vertical_align = Or(Enum("baseline", "sub", "super", "text-top", "text-bottom", "middle", "top", "bottom"), Length, Percentage, default=None)
+
+    display = Enum("none", "inline", "block", "inline-block", default=None) # TODO
+    visibility = Enum("visible", "hidden", "collapse", default=None)
+
+    white_space = Enum("normal", "pre", "nowrap", "pre-wrap", "pre-line", default=None)
+
+    word_spacing = Or(Enum("normal"), Length, Percentage, default=None)
+    letter_spacing = Or(Enum("normal"), Length, default=None)
+
+    z_index = Or(Auto, Int, default=None)
```

Comment:
At a minimum, `Styles` class needs top level docstring. Ideally, all of the properties would have help that would link the relevant page on http://www.w3schools.com/cssref/ or similar